server/plugin.go

```diff
@@ -77,3 +92,186 @@ func (p *Plugin) OnActivate() error {
 
 	return nil
 }
+
+func (p *Plugin) getOAuthConfig() (*oauth2.Config, error) {
+	config := p.getConfiguration()
+
+	if !config.enableOAuth() {
+		return nil, errors.New("please set OAuthClientID, OAuthClientSecret and EncryptionKey")
+	}
+
+	clientID := config.OAuthClientID
+	clientSecret := config.OAuthClientSecret
+	zoomUrl := config.ZoomURL
+	zoomAPIUrl := config.ZoomAPIURL
+
+	if zoomUrl == "" {
+		zoomUrl = zoomDefaultUrl
+	}
+	if zoomAPIUrl == "" {
+		zoomAPIUrl = zoomDefaultAPIUrl
+	}
+
+	authUrl := fmt.Sprintf("%v/oauth/authorize", zoomUrl)
+	tokenUrl := fmt.Sprintf("%v/oauth/token", zoomUrl)
+
+	siteUrl := *p.API.GetConfig().ServiceSettings.SiteURL
+	if siteUrl == "" {
```

Comment:
2/5 check that the pointer is not nil - it can in theory be if the customer's SiteURL is not set. Also, we should be checking it in the OnActivate and refuse to start if no valid SiteURL is set.

Reply:
Valid point, implemented the checks in both places